packages/shopping-cart/src/use-shopping-cart-manager.ts

```diff
@@ -101,6 +102,10 @@ export default function useShoppingCartManager( {
 	const isLoading = cacheStatus === 'fresh' || cacheStatus === 'fresh-pending' || ! cartKey;
 	const loadingErrorForManager = cacheStatus === 'error' ? loadingError : null;
 	const isPendingUpdate = cacheStatus !== 'valid' || ! cartKey;
+	const responseCartWithoutTempProducts = useMemo(
```

Comment:
As an alternative to this, could the hook return two different properties—one temp products and one without?

Reply:
I thought of that, which is what led to this paragraph:

> The only potential downside of this that I can think of is that it removes the possibility for optimistic updates of the shopping cart (at least, without some additional mechanism). However, since products added to the cart only include their slug and ID and neither name nor price, optimistic updates are not really very feasible anyway. If that ever becomes something desirable, and we have a means to retrieve data for not-yet-saved items, the feature can be added later.

Basically, I don't see any use in having incomplete data available to the consumer right now, so let's wait to implement that until we need it somewhere.